### ISSUE TITLE
mc config host add fails to add alias

### ISSUE THREAD

**Issue by koolhead17** [minio/mc#1605]
I did a mc pull  inside source directory & executed make, after that tried to add new host, it got added but Alias did not.

```
$ mc config host add myminio http://localhost:9000 GQAHDGEZZOPRXXLRXE49 B80dhf/2HKU4XLs9JqdjbJ8PBsWcyunu/7WBx6ip
mc: Successfully migrated /Users/koolhead17/.mc/config.json from version ‘6’ to version ‘7’.
Removing unsupported session file ‘/Users/koolhead17/.mc/session/DoImhyXU.json’ version ‘5’.
Removing unsupported session file ‘/Users/koolhead17/.mc/session/GSzgRDeZ.json’ version ‘5’.
Removing unsupported session file ‘/Users/koolhead17/.mc/session/GTmlRofO.json’ version ‘5’.
Removing unsupported session file ‘/Users/koolhead17/.mc/session/JngBvEWd.json’ version ‘5’.
Removing unsupported session file ‘/Users/koolhead17/.mc/session/LigDizkm.json’ version ‘5’.
Removing unsupported session file ‘/Users/koolhead17/.mc/session/LvGSyBKx.json’ version ‘5’.
Removing unsupported session file ‘/Users/koolhead17/.mc/session/MCLPoTZU.json’ version ‘5’.
Removing unsupported session file ‘/Users/koolhead17/.mc/session/QlkqagCh.json’ version ‘5’.
Removing unsupported session file ‘/Users/koolhead17/.mc/session/RrqUBCLz.json’ version ‘5’.
Removing unsupported session file ‘/Users/koolhead17/.mc/session/SWZnRlNr.json’ version ‘5’.
Removing unsupported session file ‘/Users/koolhead17/.mc/session/ZRUCrXVl.json’ version ‘5’.
Removing unsupported session file ‘/Users/koolhead17/.mc/session/dQAzFXmd.json’ version ‘5’.
Removing unsupported session file ‘/Users/koolhead17/.mc/session/dYIuaMYq.json’ version ‘5’.
Removing unsupported session file ‘/Users/koolhead17/.mc/session/fItyOVRC.json’ version ‘5’.
Removing unsupported session file ‘/Users/koolhead17/.mc/session/gPpiThqq.json’ version ‘5’.
Removing unsupported session file ‘/Users/koolhead17/.mc/session/gRYZmbPC.json’ version ‘5’.
Removing unsupported session file ‘/Users/koolhead17/.mc/session/hCGnMOZN.json’ version ‘5’.
Removing unsupported session file ‘/Users/koolhead17/.mc/session/hfYnPyNO.json’ version ‘5’.
Removing unsupported session file ‘/Users/koolhead17/.mc/session/nrYzoDML.json’ version ‘5’.
Removing unsupported session file ‘/Users/koolhead17/.mc/session/ojpwLWor.json’ version ‘5’.
Removing unsupported session file ‘/Users/koolhead17/.mc/session/rWLBcsRf.json’ {
        "version": "7",
        "hosts": {
                "dl": {
                        "url": "https://dl.minio.io:9000",
                        "accessKey": "",
                        "secretKey": "",
                        "api": "S3v4"
                },
                "gcs": {
                        "url": "https://storage.googleapis.com",
                        "accessKey": "YOUR-ACCESS-KEY-HERE",
                        "secretKey": "YOUR-SECRET-KEY-HERE",
                        "api": "S3v2"
                },
                "local": {
                        "url": "http://localhost:9000",
                        "accessKey": "",
                        "secretKey": "",
                        "api": "S3v4"
                },
                "play": {
                        "url": "https://play.minio.io:9000",
                        "accessKey": "",
...skipping...
{
        "version": "7",
        "hosts": {
                "dl": {
                        "url": "https://dl.minio.io:9000",
                        "accessKey": "",
                        "secretKey": "",
                        "api": "S3v4"
                },
                "gcs": {
                        "url": "https://storage.googleapis.com",
                        "accessKey": "YOUR-ACCESS-KEY-HERE",
                        "secretKey": "YOUR-SECRET-KEY-HERE",
                        "api": "S3v2"
                },
                "local": {
version ‘5’.
Removing unsupported session file ‘/Users/koolhead17/.mc/session/suRuMrCN.json’ version ‘5’.
Removing unsupported session file ‘/Users/koolhead17/.mc/session/sxIGottl.json’ version ‘5’.
Removing unsupported session file ‘/Users/koolhead17/.mc/session/uIhYFwcc.json’ version ‘5’.
Removing unsupported session file ‘/Users/koolhead17/.mc/session/uwAatREI.json’ version ‘5’.
Added ‘myminio’ successfully.
$
$ mc mirror myminio/books s3/ferenginar
mc: Successfully migrated /Users/koolhead17/.mc/config.json from version ‘6’ to version ‘7’.
...ystem.pdf: 874.20 KB / 874.20 KB  ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓  100.00 % 90.05 MB/s 0$ mc ls s3/ferenginar
mc: <ERROR> Unable to stat ‘s3/ferenginar’. The AWS Access Key Id you provided does not exist in our records.
$ less ~/.mc/config.json
$ mc config host list
s3: https://s3.amazonaws.com | <- YOUR-ACCESS-KEY-HERE, |  YOUR-SECRET-KEY-HERE, |  S3v4
dl: https://dl.minio.io:9000
gcs: https://storage.googleapis.com | <- YOUR-ACCESS-KEY-HERE, |  YOUR-SECRET-KEY-HERE, |  S3v2
local: http://localhost:9000
play: https://play.minio.io:9000
player: https://play.minio.io:9002 | <- Q3AM3UQ867SPQQA43P2F, |  zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG, |  S3v4
$ mc config host list
s3: https://s3.amazonaws.com | <- YOUR-ACCESS-KEY-HERE, |  YOUR-SECRET-KEY-HERE, |  S3v4
dl: https://dl.minio.io:9000
gcs: https://storage.googleapis.com | <- YOUR-ACCESS-KEY-HERE, |  YOUR-SECRET-KEY-HERE, |  S3v2
local: http://localhost:9000
play: https://play.minio.io:9000
player: https://play.minio.io:9002 | <- Q3AM3UQ867SPQQA43P2F, |  zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG, |  S3v4
$ mc config host add myminio http://localhost:9000 GQAHDGEZZOPRXXLRXE49 B80dhf/2HKU4XLs9JqdjbJ8PBsWcyunu/7WBx6ip
Added ‘myminio’ successfully.

```

**Comment by harshavardhana** [comment]
> local: http://localhost:9000
> play: https://play.minio.io:9000
> player: https://play.minio.io:9002 | <- Q3AM3UQ867SPQQA43P2F, |  zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG, |  S3v4
> $ mc config host list
> s3: https://s3.amazonaws.com | <- YOUR-ACCESS-KEY-HERE, |  YOUR-SECRET-KEY-HERE, |  S3v4
> dl: https://dl.minio.io:9000
> gcs: https://storage.googleapis.com | <- YOUR-ACCESS-KEY-HERE, |  YOUR-SECRET-KEY-HERE, |  S3v2
> local: http://localhost:9000
> play: https://play.minio.io:9000
> player: https://play.minio.io:9002 | <- Q3AM3UQ867SPQQA43P2F, |  zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG, |  S3v4
> $ mc config host add myminio http://localhost:9000 GQAHDGEZZOPRXXLRXE49 B80dhf/2HKU4XLs9JqdjbJ8PBsWcyunu/7WBx6ip
> Added ‘myminio’ successfully.

This is a race condition, the problem is that config files were being migrated from their previous version to new version, but at the same time you were adding an alias which was overwritten by default values from the migration code. 

I do not know if there is an easier way to fix this and it is quite rare, it has to hit the above conditions i guess :-)

**Comment by harshavardhana** [comment]
This is reproducible, whenever there is a config migrate. There is a small window where the config values are not available.

**Comment by harshavardhana** [comment]
This is fixed now by #1666
